log4j-core/src/main/java/org/apache/logging/log4j/core/config/DefaultConfiguration.java

```diff
@@ -49,4 +61,60 @@ public DefaultConfiguration() {
 
     @Override
     protected void doConfigure() {}
+
+    static Layout<? extends String> createDefaultLayout() {
+        return new DefaultLayout();
+    }
+
+    /**
+     * A simple layout used only by {@link DefaultConfiguration}
+     * <p>
+     *   This layout allows to create applications that don't contain {@link org.apache.logging.log4j.core.layout.PatternLayout}
+     *   and all its patterns, e.g. GraalVM applications.
+     * </p>
+     */
+    private static final class DefaultLayout extends AbstractLayout<String> {
```

Comment:
Is there any way we can reuse logic from `StatusData#getFormattedStatus()` or `SimpleLogger`? That is, instead of creating yet another primitive event formatter, maybe we can reuse something already there in the `log4j-api`?

Reply:
Good call!

I went for `StatusData` in https://github.com/apache/logging-log4j2/pull/3118/commits/bf892e169e273d3e9c4e4812e4ffac4178460a3d
This slightly changes the format and removes the name of the logger, but I really doubt there is a real time situation were `DefaultConfiguration` is used. Logging is usually configured **before** a JVM starts new threads, so the concurrency problem in `LoggerContext` is mostly a worst case scenario.